src/microbe_stage/organelle_components/StorageComponent.cs

```diff
@@ -29,3 +29,22 @@ public void Check(string name)
         }
     }
 }
+
+[JSONDynamicTypeAllowed]
+public class StorageComponentUpgrades : IComponentSpecificUpgrades
+{
+    public StorageComponentUpgrades(bool isSpecialized, Compound specialization)
+    {
+        Specialization = specialization;
+        IsSpecialized = isSpecialized;
+    }
+
+    public Compound Specialization { get; set; }
```

Comment:
Is there a reason for using two variables? I'd just make one property `Compound? SpecializedFor` and make it so that null means no specialization and then a valid compound value is specialization for that compound.